### PR TITLE
Route to /blog on "All blog posts" navigation

### PR DESCRIPTION
**Pull request by klarzynskik**
Currently, the "All blog posts" button on the particular blog post was using the `router.back()`, which depending on the context might not work at all (e.g. when visiting from twitter with `target="_blank"`) or route to wrong page (e.g. back to google results):

https://user-images.githubusercontent.com/47242672/145561455-8638723f-1055-49df-9a66-3e40244dc3b5.mov

I think it would be better if it takes the User to the `/blog` directly, and the same is for the `404`, which was redirecting to the main page instead.

### CODE REVIEW

File: pages/404.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ const NotFound: NextPage<{ posts: Post[] }> = ({ posts }) => (
       </Heading>
       <Box as="p" textAlign="left" padding="6">
         Sorry we can’t find that page.{" "}
-        <ChakraLink href="/" color="primary.300" textDecoration="underline">
+        <ChakraLink href="/blog" color="primary.300" textDecoration="underline">
```
Comment (mattpocock) [inline pages/404.tsx:38]: @klarzynskik Could you use `next/link` for this? This will make the navigation smoother.

**Review by mattpocock — CHANGES_REQUESTED** [review]
Thanks so much for the contribution! We'd noticed this bug ourselves and were struggling to get round to it.

File: pages/[slug].tsx [hunk 2]
```diff
@@ -53,24 +49,23 @@ const PostPage: React.FC<{
         >
           <Button
             as="a"
-            onClick={() => {
-              router.back();
-            }}
+            href="/blog"
```
Comment (mattpocock) [inline pages/[slug].tsx:52]: Same here - use `next/link` as a wrapper around this button.

File: pages/[slug].tsx [hunk 3]
```diff
@@ -53,24 +49,23 @@ const PostPage: React.FC<{
...
             marginBottom="8"
             cursor="pointer"
             textDecoration="none"
             leftIcon={<ArrowBackIcon />}
           >
             All blog posts
           </Button>
```
Comment (farskid) [inline pages/[slug].tsx:59]: Let's use `next/link` and `ChakraLink` here. now that it's a link, we don't need the button anymore.

```jsx
<NextLink href="/blog" passHref>
  <ChakraLink>All blog posts</ChakraLink>
</NextLink>
```

File: pages/404.tsx (hunk 1 above)
Comment (farskid) [reply pages/404.tsx:38]: Let's use both `next/link` and `ChakraLink` here.

**Review by farskid — CHANGES_REQUESTED** [review]
Small changes requested, otherwise LG! thank you ❤️

File: pages/[slug].tsx (hunk 3 above)
Comment (mattpocock) [reply pages/[slug].tsx:59]: `<Button as="a" />` is also fine in my book, have used that trick a few times.

Comment (klarzynskik) [reply pages/[slug].tsx]: I went with what @farskid suggested (the `/blog` is not needed after using `Link`, with it correctly resolves to `basePath` now), let me know if you want it to be back with Button (the design has changed a bit after that) 😅

Comment (farskid) [reply pages/[slug].tsx]: @mattpocock ita fine but we want a combination of NextLink and ChakraLink like other places on the code.

Comment (mattpocock) [reply pages/[slug].tsx]: But the Button will look better? I.e. it will look like it does now, but will act like a link.

Comment (mattpocock) [reply pages/[slug].tsx]: @klarzynskik You're right, `/` is correct because of the basePath

Comment (mattpocock) [reply pages/[slug].tsx]: Also, consider my comments above nit-level - happy to add my approval and step away

**Review by mattpocock — APPROVED** [review]

File: pages/[slug].tsx (hunk 3 above)
Comment (farskid) [reply pages/[slug].tsx]: @mattpocock Actually we can have `<ChakraLink as={Button} />` to make look like a button

**Review by farskid — CHANGES_REQUESTED** [review]
@klarzynskik Sorry for having nits.
1. Let's remove unused imports and codes from `[slug].tsx` now that the router isn't used anymore
2. Let's add `as={Button}` to the back link to make it look like a button and use `<ArrowBackIcon />` behind the text.

**Comment by klarzynskik** [comment]
Thanks @farskid!

Ad.1 I think I have removed the router and other unused code in https://github.com/statelyai/eng-blog/pull/56/commits/9b070fab428d7f66dc68aef1628c5c8499316fca already, do you see anything else that needs to be removed?
Ad.2 I've adjusted that in https://github.com/statelyai/eng-blog/pull/56/commits/041f6eabb1b1e47a46975690c8bb458d9d1b7d7d, I've used additional CSS to make the icon vertically aligned with the text. Let me know your thoughts.

**Comment by farskid** [comment]
> Thanks @farskid!
> 
> Ad.1 I think I have removed the router and other unused code in [9b070fa](https://github.com/statelyai/eng-blog/commit/9b070fab428d7f66dc68aef1628c5c8499316fca) already, do you see anything else that needs to be removed? Ad.2 I've adjusted that in [7042cb6](https://github.com/statelyai/eng-blog/commit/7042cb627b1dde783fc6d25419a155b0316010e5). Let me know your thoughts.

So I made a mistake about `<Link as={Button}` 😞 it should be `<NextLink><Button as="a"></Button></NextLink>` for accessibility and valid HTML.

We need a small tweak to styles to keep the same look after changing to a link. The following snippet will do. Sorry for being picky and the mistake about right components.
```jsx
<Link href="/" passHref>
  <Button as="a" textDecoration="none" marginBottom="8">
    {<ArrowBackIcon marginRight="2" />} All blog posts
  </Button>
</Link>
```

**Comment by klarzynskik** [comment]
@farskid no worries, I hope it will be okay now 🙌🏼

**Comment by farskid** [comment]
> @farskid no worries, I hope it will be okay now 🙌🏼

Perfect! Thank you. Ready for merge.

**Pull request merged**